### PR TITLE
Move unboxed float literals to stable

### PR DESCRIPTION
**Pull request by alanechang**
#2062 and #2071 added proper middle/backend support for unboxed float literals which motivates moving it out of alpha.

### CODE REVIEW

**Review by goldfirere — APPROVED** [review]

**Comment by goldfirere** [comment]
I can sometimes be rather gung-ho about changes like this, so would also like a second opinion from @ccasin.

**Comment by ccasin** [comment]
> I can sometimes be rather gung-ho about changes like this, so would also like a second opinion from @ccasin.

Seems fine, but maybe now would be a good moment to update `ocaml/testsuite/tests/typing-layouts-float64/unboxed-floats.ml` to use unboxed literals rather than `of_float 3.14` everywhere, for a little bonus confidence in their dynamic behavior?

**Comment by alanechang** [comment]
I updated all tests in `ocaml/testsuite/tests/typing-layouts-float64/` to use unboxed float literals. 

`unboxed_floats_disabled.compilers.reference` is the only test output that changed. Also had to mark all the tests with `* flambda2` due to it being the only middle end with unboxed literal support.

File: ocaml/testsuite/tests/typing-layouts-float64/unboxed_floats.ml [hunk 1]
```diff
@@ -1,22 +1,23 @@
 (* TEST
+   * flambda2
```
Comment (goldfirere) [inline ocaml/testsuite/tests/typing-layouts-float64/unboxed_floats.ml:2]: Why was this change necessary? I remember hearing that all the backends should be working.

**Review by goldfirere — COMMENTED** [review]
Just one question. If that's resolved, we should merge.

File: ocaml/testsuite/tests/typing-layouts-float64/unboxed_floats.ml (hunk 1 above)
Comment (ccasin) [reply ocaml/testsuite/tests/typing-layouts-float64/unboxed_floats.ml:2]: That was true before literals, but I assume this is necessary because the literals themselves have only been implemented for flambda2?  I am sad about that, but not sad enough to slow this down.

Comment (alanechang) [reply ocaml/testsuite/tests/typing-layouts-float64/unboxed_floats.ml:2]: Yes, unboxed literals are currently only supported on flambda2.

The first implementation of the feature was compatible with all middle/backends when it compiles to calling unbox on a boxed constant. But when #2062 and #2071 changed the implementation to compile directly to an unboxed constant, closure and flambda1 support were dropped.
